src/optimizer/join_order/cardinality_estimator.cpp

```diff
@@ -432,11 +432,13 @@ void CardinalityEstimator::UpdateTotalDomains(JoinNode &node, LogicalOperator &o
 			}
 		}
 
+		bool have_stats = false;
 		if (catalog_table && actual_binding != relation_column_to_original_column.end()) {
```

Comment:
In order to avoid regression on some IMDB queries, I've left this conditional the same. It should be `if (get_updated && ...`, but then we have the following regressions on the IMDB plan cost benchmark (comparing exact cardinality counts, not with the 20% threshold).  The bug is that during the process of initializing relation columns with certain base cardinality counts, if the get operator is not updated for iteration i+1, but the catalog table exists from from iteration i of the for loop on line `409` then we still get the statistics. 

I checked to see if changing the conditional would fix the [logical column index out of range error](https://github.com/duckdb/duckdb/issues/7905), but it doesn't. I think for now we should keep it. Maybe I can add a TODO: for future reference?

14 improvements (3 of which are over 20%)
25 regressions (9 of which are over 20%)

exact counts
```

RUNNING BENCHMARK QUERIES

====================================================
===========     IMPROVEMENTS DETECTED     ==========
====================================================

Query: 08d
Old cost: 35528373
New cost: 34697580

Query: 18a
Old cost: 1408455
New cost: 1405430

Query: 21a
Old cost: 14594
New cost: 14249

Query: 21b
Old cost: 11716
New cost: 11491

Query: 21c
Old cost: 15938
New cost: 15369

Query: 24a
Old cost: 2848278
New cost: 2187180

Query: 27a
Old cost: 13647
New cost: 13180

Query: 27b
Old cost: 12374
New cost: 12127

Query: 28a
Old cost: 1506654
New cost: 1499746

Query: 28c
Old cost: 1858987
New cost: 1845036

Query: 29a
Old cost: 9720
New cost: 8090

Query: 29c
Old cost: 115933
New cost: 98864

Query: 30a
Old cost: 652211
New cost: 634262

Query: 31a
Old cost: 1877939
New cost: 829467

Query: 31c
Old cost: 1940553
New cost: 1364844

====================================================
===========     REGRESSIONS DETECTED     ===========
====================================================

Query: 08a
Old cost: 49980
New cost: 57356

Query: 08c
Old cost: 41542705
New cost: 42876518

Query: 09a
Old cost: 821906
New cost: 850137

Query: 09b
Old cost: 223418
New cost: 292761

Query: 09c
Old cost: 2045156
New cost: 2390904

Query: 11a
Old cost: 12491
New cost: 12839

Query: 17b
Old cost: 8133770
New cost: 8230016

Query: 17c
Old cost: 8032994
New cost: 8179628

Query: 19a
Old cost: 1032531
New cost: 1067938

Query: 19b
Old cost: 155
New cost: 184

Query: 19c
Old cost: 2984681
New cost: 3102516

Query: 20a
Old cost: 1742754
New cost: 2369449

Query: 20b
Old cost: 574592
New cost: 703444

Query: 20c
Old cost: 552481
New cost: 865771

Query: 22a
Old cost: 742913
New cost: 771735

Query: 22b
Old cost: 370230
New cost: 400723

Query: 22c
Old cost: 3570607
New cost: 3585770

Query: 24b
Old cost: 689
New cost: 1623

Query: 26a
Old cost: 305499
New cost: 391769

Query: 26b
Old cost: 27932
New cost: 34072

Query: 26c
Old cost: 1084225
New cost: 1358910

Query: 28b
Old cost: 327545
New cost: 342001

Query: 30b
Old cost: 618
New cost: 532849

Query: 33a
Old cost: 26784
New cost: 1640472

Query: 33b
Old cost: 2589
New cost: 83917

Query: 33c
Old cost: 38790
New cost: 1842471
```